### PR TITLE
Allow CMake to select a default internal generator when the user selects "__unspec__" for the kit

### PR DESCRIPTION
**Pull request by snehara99**
## This change addresses item #3405 

CMake file api supports not passing in a generator in versions >= 3.15.0. I added a check for that version and using the file api in order to not pass in Ninja as a preferred generator when the user selects "__unspec__" as a kit and doesn't define any generator/preferredGenerators.

### CODE REVIEW

File: src/drivers/cmakeFileApiDriver.ts [hunk 1]
```diff
@@ -38,7 +38,7 @@ const localize: nls.LocalizeFunc = nls.loadMessageBundle();
 
 const log = logging.createLogger('cmakefileapi-driver');
 /**
- * The CMake driver with FileApi of CMake >= 3.15.0
+ * The CMake driver with FileApi of CMake >= 3.14.0
```
Comment (gcampbell-msft) [inline src/drivers/cmakeFileApiDriver.ts:41]: Why is this version being changed?

**Review by gcampbell-msft — COMMENTED** [review]
Overall this looks good, assuming we confirm with testing that it works as expected and our tests pass, and I have one comment that needs addressing.

File: src/drivers/cmakeFileApiDriver.ts (hunk 1 above)
Comment (snehara99) [reply src/drivers/cmakeFileApiDriver.ts:41]: The version isn't changed. I think the version in the comment was incorrect for some reason. This is the actual version we check (unmodified) 
image

Comment (gcampbell-msft) [reply src/drivers/cmakeFileApiDriver.ts:41]: Ahh, thanks for pointing that out, thanks! 👍

**Comment by snehara99** [comment]
> Overall this looks good, assuming we confirm with testing that it works as expected and our tests pass, and I have one comment that needs addressing.

I've tested it a decent amount. I think it should be fine

**Review by gcampbell-msft — APPROVED** [review]

**Pull request merged**